test/unittest/dds/participant/ParticipantTests.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+
+TEST(ParticipantTest, DomainParticipantFactoryGetInstance)
+{
+    DomainParticipantFactory* factory = DomainParticipantFactory::get_instance();
+
+    ASSERT_NE(factory, nullptr);
```

Comment:
I would also check that a second call gives the same result. This is a singleton, right?